src/DrWatson.jl

```diff
@@ -45,6 +45,19 @@ using Requires
 function __init__()
     @require DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0" begin
         include("result_collection.jl")
+        @require JLD2 = "033835bb-8acc-5ee8-8aae-3f567f8a3819" begin
+            using .JLD2
+            # This allows us to use the much faster jldopen in collect_results
+            function load_jld2(f, path)
+                @debug "Opening $(path) with jldopen."
+                jldopen(path) do data
+                    return f(data)
+                end
+            end
+            ext_register[".jld2"] = load_jld2
+            # JLD2 doesn't define keytype, but only supports Strings anyway.
+            Base.keytype(f::JLD2.JLDFile) = String
+        end
```

Comment:
Hm, something to keep in mind here. This code becomes available if a user has `using JLD2` in the start of their script. 

I, for example, do not do this. I don't `using JLD2`, even though I save with JLD2. As long as JLD2 is installed, FileIO does not require you to actually use the package. So perhaps we need to think of a different design that doesn't go this Requires route. Maybe we can have and `if` clause inside `collect_results`, so that if the file endings are jld2 it uses your faster method?

Reply:
I did this intentionally such that JLD2 doesn't become a hard dependency. It would actually be better to first check if JLD2 is available and if yes, load it, but I do not know the canonical way for this in Julia (but I just need to find the time to check how FileIO does it).

Right now, if JLD2 isn't loaded, the code will fall back to `wload` for the first file. This triggers JLD2 to be loaded via FileIO. From the second file onward, the fast path via `jldopen` is available and will used.

Reply:
Huh, are you sure about that? Have you actually tested it that it works this way? I don't know for sure.

Reply:
Yes, I tested this in my workflow. But I didn't write a test for it because that would require a reorganisation of `tests/update_results_tests.jl`. I'd need to run that test before the `using JLD2` statement at the top.

What FileIO does is call `Base.require` inside a `try` block. The code is a bit obscure because it's all wrapped in macros, but I think this would also be a feasible route to take.

Reply:
OOOooh okay, thanks for being thorough with this, one can never be certain haha! 

Okay, then I'll let Jonas review and merge this, given his expertise on the subject. All criticism I had to give I gave!

Reply:
Great, thank you!